Native application execution failure with @Retry and overridden methods

### Describe the bug

Hello,

my native application fails to start with the following logs:
```
2025-05-16 09:03:09,069 ERROR [io.qua.run.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:80)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:51)
        at MyApp.main(MyApp.java:17)
Caused by: jakarta.enterprise.inject.spi.DeploymentException: java.lang.NoSuchMethodException: AbstractService.myPublicMethod()
        at io.quarkus.smallrye.faulttolerance.runtime.SmallRyeFaultToleranceRecorder.createFaultToleranceOperation(SmallRyeFaultToleranceRecorder.java:44)
        at io.quarkus.runner.recorded.SmallRyeFaultToleranceProcessor$processFaultToleranceAnnotations1222864980.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.SmallRyeFaultToleranceProcessor$processFaultToleranceAnnotations1222864980.deploy(Unknown Source)
        ... 6 more
Caused by: java.lang.NoSuchMethodException: AbstractService.myPublicMethod()
        at java.base@21.0.7/java.lang.Class.checkMethod(DynamicHub.java:1078)
        at java.base@21.0.7/java.lang.Class.getDeclaredMethod(DynamicHub.java:1168)
        at io.smallrye.faulttolerance.autoconfig.MethodDescriptor.reflect(MethodDescriptor.java:28)
        at io.quarkus.smallrye.faulttolerance.runtime.SmallRyeFaultToleranceRecorder.createFaultToleranceOperation(SmallRyeFaultToleranceRecorder.java:31)
        ... 8 more
```

This is caused by the use of @Retry in a class (abstract class in my case but it is probably not important) at class level and a method overridden in another class extending the class with the @Retry annotation.

Thanks.

### Expected behavior

The application start.

### Actual behavior

The application fails to start

### How to Reproduce?

[code-with-quarkus-native-retry-2.zip](https://github.com/user-attachments/files/20239664/code-with-quarkus-native-retry-2.zip)

- build a native executable (with Docker mvn install -Dnative -DskipTests -Dquarkus.native.container-build=true)
- run the native app: ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner

### Output of `uname -a` or `ver`

Ubuntu 24.04.2 LTS

### Output of `java -version`

openjdk version "21.0.6" 2025-01-21

### Mandrel or GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.22.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.7

### Additional information

_No response_

--- a/extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java
+++ b/extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java
@@ -315,6 +315,8 @@ public class SmallRyeFaultToleranceProcessor {
                     if (ftMethod.isLegitimate()) {
                         ftMethods.add(ftMethod);
 
+                        reflectiveMethod.produce(new ReflectiveMethodBuildItem("fault tolerance method", method));
+
                         if (annotationStore.hasAnnotation(method, DotNames.ASYNCHRONOUS)
                                 && annotationStore.hasAnnotation(method, DotNames.ASYNCHRONOUS_NON_BLOCKING)) {
                             exceptions.add(new DefinitionException(

